repositories/pokemon_repo.go

```diff
@@ -48,3 +54,85 @@ func (pr PokemonRepo) WritePokemonCsvFile(pokemons []models.Pokemon) error {
 	}
 	return nil
 }
+
+func (pr PokemonRepo) GetPokemonsFromCSVConcurrently(idType string, items int, itemsPerWorker int) ([]models.Pokemon, error) {
+	numberOfWorkers := items / itemsPerWorker
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+	defer cancel()
+	row := make(chan []string)
+	pokemon := make(chan models.Pokemon)
```

Comment:
i would suggest making this channel a buffered channel. The way you have it here you will only be able to insert and take one item at a time, which will make all of your threads wait on each other, by making it buffered you will be able to insert items from multiple threads at the same time with no issue, which will reduce the processing time and will actually achieve what we're looking for.